src/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php

```diff
@@ -243,7 +243,13 @@ public function processSingleLineArray($phpcsFile, $stackPtr, $arrayStart, $arra
                 if ($fix === true) {
                     $phpcsFile->fixer->beginChangeset();
                     $phpcsFile->fixer->addNewline($arrayStart);
-                    $phpcsFile->fixer->addNewlineBefore($arrayEnd);
+
+                    if ($tokens[($arrayEnd - 1)]['code'] === T_WHITESPACE) {
```

Comment:
The syntax used in some other sniff seems to be: 
```
$prevToken = $phpcsFile->findPrevious(T_WHITESPACE, ($token - 1), null, true);

for ($i = ($prevToken + 1); $i < $token; $i++) {
    $phpcsFile->fixer->replaceToken($i, '');
}

$phpcsFile->fixer->addNewLineBefore($token);
```

Reply:
I don't understand how that would apply here, please elaborate.

Reply:
In some other sniff of this project, when adding a new line, the fixer
- find the previous token, excluding whitespace
- replace every whitespace between the previous token and the token to fix
- add a new line.

Here, it would be
```
$token = $tokens[($arrayEnd - 1)];
$prevToken = $phpcsFile->findPrevious(T_WHITESPACE, ($token - 1), null, true);

for ($i = ($prevToken + 1); $i < $token; $i++) {
    $phpcsFile->fixer->replaceToken($i, '');
}

$phpcsFile->fixer->addNewLineBefore($token);
```

But I then saw here https://github.com/squizlabs/PHP_CodeSniffer/pull/2812/files#r365493493 that the syntax 'if previous is whitespace, replace by a new line, if not add a new line` is already used in this sniff. 
So it seems there is no standard about "How to do it".

So unless a syntax is preferred there is nothing to change

Reply:
As far as I understand it, it won't replace "every whitespace", but everything. So, if you had a comment there, it would be gone. Sorry but this code doesn't make sense to me.

Anyway, I believe there isn't such a thing as a "standard syntax" for any fixes.

Reply:
FYI : `$phpcsFile->findPrevious(T_WHITESPACE, ($token - 1), null, true);` will find the first token which is not a whitespace. The param `true` is used for `$excluded`.

> Anyway, I believe there isn't such a thing as a "standard syntax" for any fixes.

Yes so this can be resolved.